### PR TITLE
add git hook to make sure assets/$version has been gitignored

### PR DESCRIPTION
**Pull request by nanaian**
This will hopefully dissuade any modders from trying to commit vanilla assets to their repos

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,7 +26,6 @@ build.ninja
 *.z64
 *.Yay0
 *.msg.h
-*.bin
```
Comment (nanaian) [inline .gitignore:29]: This is to enable people to override `.bin` files with an asset stack. We already ignore the dump asset directories

File: .gitignore [hunk 2]
```diff
@@ -41,7 +40,12 @@ build/
 /tools/permuter_settings.toml
 
 # Assets
-/assets/*
```
Comment (nanaian) [inline .gitignore:44]: This is to make it so people hopefully won't have to go modifying gitignore when adding new asset directories

**Comment by nanaian** [comment]
Overall it would be better if we checked to see if the commit _contains_ any assets, since force-adding is still possible, but this is probably good enough & isn't an expensive check to make on commit

**Comment by Mastergatto** [comment]
Hi, may I suggest to also add *.n64 and *.v64 to .gitignore file? Just to be 100% sure, although this may be a thing that is unlikely to occur, if it ever happens.

**Pull request merged**